src/pages/index.js

```diff
@@ -97,19 +160,66 @@ function createCard(item) {
     {
       name: item.name,
       link: item.link,
+      id: item.id,
+      isLiked: item.isLiked,
     },
     ".card-template",
-    openImageModal
+    openImageModal,
+    openDeleteModal,
+    handleCardLike
   );
   return cardObj.generateCard();
 }
 
+function handleCardLike(card) {
+  api
+    .likeCard({
+      cardId: card.getID(),
+      likeBool: card.getIsLiked(),
+    })
+    .then((res) => {
+      card.setIsLiked(res.isLiked);
+    });
+}
+
+// on load render preexisting cards
+api.renderCardsValidator().then((res) => {
```

Comment:
In `res` you get cards and the user info


Please, use them here rather than call `api.getUserInformation()` again in line 203



```js

Promise.all([getUserInfo(), getCards()])
// destructure the response
  .then(([userData, cards]) => {
      // set all the data
  })
  .catch(err => {
    // catch possible errors
  });

```